orte/mca/rml/ofi/rml_ofi.h

```diff
@@ -0,0 +1,193 @@
+
+#include "orte_config.h"
+
+#include "opal/dss/dss_types.h"
+#include "opal/mca/event/event.h"
+#include "opal/mca/pmix/pmix.h"
+#include "orte/mca/rml/base/base.h"
+
+#include <rdma/fabric.h>
+#include <rdma/fi_cm.h>
+#include <rdma/fi_domain.h>
+#include <rdma/fi_endpoint.h>
+#include <rdma/fi_errno.h>
+#include <rdma/fi_tagged.h>
+
+#include "rml_ofi_request.h"
+
+//[A]
+/** the maximum open conduit - assuming system will have no more than 20 transports*/
+#define MAX_CONDUIT  20
+
+/** The OPAL key values  **/
+/* (char*)  ofi socket address (type IN) of the node process is running on */
+#define OPAL_RML_OFI_FI_SOCKADDR_IN                "rml.ofi.fisockaddrin"   
+/* (char*)  ofi socket address (type PSM) of the node process is running on */  
+#define OPAL_RML_OFI_FI_ADDR_PSMX                  "rml.ofi.fiaddrpsmx"       
+
+// MULTI_BUF_SIZE_FACTOR defines how large the multi recv buffer will be.
+// In order to use FI_MULTI_RECV feature efficiently, we need to have a 
+// large recv buffer so that we don't need to repost the buffer often to 
+// get the remaining data when the buffer is full
+#define MULTI_BUF_SIZE_FACTOR 128
+#define MIN_MULTI_BUF_SIZE (1024 * 1024)
+
+#define CLOSE_FID(fd)			\
+	ret = 0;			\
+	do {				\
+		if ((fd)) {					\
+			ret = fi_close(&(fd)->fid);	\
+			fd = NULL;				\
+			if (ret)				\
+    		        {					\
+			    opal_output_verbose(10,orte_rml_base_framework.framework_output, 	\
+                                                " %s - fi_close failed with error- %d",		\
+                                                   ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),ret);	\
+		        }					\
+		}						\
+	} while (0);
+
+
+#define RML_OFI_RETRY_UNTIL_DONE(FUNC)         \
+    do {                                       \
+        do {                                   \
+            ret = FUNC;                        \
+            if(OPAL_LIKELY(0 == ret)) {break;} \
+        } while(-FI_EAGAIN == ret);            \
+    } while(0);
+
+BEGIN_C_DECLS
+
+
+/** This structure will hold the ep and all ofi objects for each transport
+and also the corresponding fi_info
+**/
+typedef struct {
+
+    /** conduit ID **/
+    uint8_t conduit_id;
+
+    /** fi_info for this transport */
+    struct fi_info *fabric_info;
+
+    /** Fabric Domain handle */
+    struct fid_fabric *fabric;
+
+    /** Access Domain handle */
+    struct fid_domain *domain;
+
+    /** Address vector handle */
+    struct fid_av *av;
+
+    /** Completion queue handle */
+    struct fid_cq *cq;
+
+    /** Endpoint to communicate on */
+    struct fid_ep *ep;
+
+    /** Endpoint name */
+    char ep_name[FI_NAME_MAX];
+
+    /** Endpoint name length */
+    size_t epnamelen;
+
+    /** OFI memory region */
+    struct fid_mr *mr_multi_recv;
+
+	/** buffer for tx and rx */
+    void *rxbuf;
+ 
+    uint64_t rxbuf_size;
+	
+	/* event,fd associated with the cq */ 
+	int fd;
+
+	/*event associated with progress fn */
+	opal_event_t progress_event;
```

Comment:
There is inconsistent indenting in this file -- did some tabs accidentally get used?